lib/computer.rb

```diff
@@ -8,29 +8,33 @@ class Computer
   def initialize(grid, ships_list)
     @grid = grid
     @ships_list = ships_list
-    @create_ship = CreateShip.new
   end
 
-  def place_ship
-    while @ships_list.ships.size > 0
-      random_ship = select_random_ship
-      position = valid_position(select_random_position, random_ship.length)
-      random_ship.register_position(position[:x], position[:y], position[:orientation])
-      @grid.add_ship(random_ship)
-    end
+  def ship_placement
+    random_ship = generate_ship
+    position = generate_position(random_ship.length)
+    random_ship.register_position(position[:x], position[:y], position[:orientation])
+    random_ship
   end
 
   private
 
-  def valid_position(position, ship_length)
-    validation_result = @grid.validate_placement(position, ship_length)
+  def generate_ship
+    select_random_ship
+  end
+
+  def generate_position(ship_length)
+    valid_position(ship_length)
+  end
+
+  def valid_position(ship_length)
```

Comment:
I feel like this method has two responsibility, get a random move and validate this random move. Wouldn't it be easier to read if it was split into two different methods? Something like that:

``` ruby
def get_next_position(ship)
    position = random_position(ship)
    return position if available?(position)
    get_next_position(ship)
end
```